src/util/newconfig/ConfigDefinition.cpp

```diff
@@ -0,0 +1,213 @@
+     {"dos_guard.max_requests", ConfigValue{ConfigType::Integer}.defaultValue(20)},
+     {"dos_guard.sweep_interval", ConfigValue{ConfigType::Double}.defaultValue(1.0)},
+     {"cache.peers.[].ip", Array{ConfigValue{ConfigType::String}}},
+     {"cache.peers.[].port", Array{ConfigValue{ConfigType::String}}},
+     {"server.ip", ConfigValue{ConfigType::String}},
+     {"server.port", ConfigValue{ConfigType::Integer}},
+     {"server.max_queue_size", ConfigValue{ConfigType::Integer}.defaultValue(0)},
+     {"server.local_admin", ConfigValue{ConfigType::Boolean}.optional()},
+     {"prometheus.enabled", ConfigValue{ConfigType::Boolean}.defaultValue(true)},
+     {"prometheus.compress_reply", ConfigValue{ConfigType::Boolean}.defaultValue(true)},
+     {"io_threads", ConfigValue{ConfigType::Integer}.defaultValue(2)},
+     {"cache.num_diffs", ConfigValue{ConfigType::Integer}.defaultValue(32)},
+     {"cache.num_markers", ConfigValue{ConfigType::Integer}.defaultValue(48)},
+     {"cache.page_fetch_size", ConfigValue{ConfigType::Integer}.defaultValue(512)},
+     {"cache.load", ConfigValue{ConfigType::String}.defaultValue("async")},
+     {"log_channels.[].channel", Array{ConfigValue{ConfigType::String}.optional()}},
+     {"log_channels.[].log_level", Array{ConfigValue{ConfigType::String}.optional()}},
+     {"log_level", ConfigValue{ConfigType::String}.defaultValue("info")},
+     {"log_format",
+      ConfigValue{ConfigType::String}.defaultValue(
+          R"(%TimeStamp% (%SourceLocation%) [%ThreadID%] %Channel%:%Severity% %Message%)"
+      )},
+     {"log_to_console", ConfigValue{ConfigType::Boolean}.defaultValue(false)},
+     {"log_directory", ConfigValue{ConfigType::String}.optional()},
+     {"log_rotation_size", ConfigValue{ConfigType::Integer}.defaultValue(2048)},
+     {"log_directory_max_size", ConfigValue{ConfigType::Integer}.defaultValue(50 * 1024)},
+     {"log_rotation_hour_interval", ConfigValue{ConfigType::Integer}.defaultValue(12)},
+     {"log_tag_style", ConfigValue{ConfigType::String}.defaultValue("uint")},
+     {"extractor_threads", ConfigValue{ConfigType::Integer}.defaultValue(2)},
+     {"read_only", ConfigValue{ConfigType::Boolean}.defaultValue(false)},
+     {"txn_threshold", ConfigValue{ConfigType::Integer}.defaultValue(0)},
+     {"start_sequence", ConfigValue{ConfigType::String}.optional()},
+     {"finish_sequence", ConfigValue{ConfigType::String}.optional()},
+     {"ssl_cert_file", ConfigValue{ConfigType::String}.optional()},
+     {"ssl_key_file", ConfigValue{ConfigType::String}.optional()},
+     {"api_version.min", ConfigValue{ConfigType::Integer}},
+     {"api_version.max", ConfigValue{ConfigType::Integer}}}
+};
+
+ObjectView
+ClioConfigDefinition::getObject(std::string_view prefix, std::optional<std::size_t> idx) const
+{
+    std::string prefixWithDot = std::string(prefix) + ".";
+    for (auto const& [mapKey, mapVal] : map_) {
+        if (mapKey.starts_with(prefixWithDot))
```

Comment:
You might want to add idx is nullopt here? **if (mapKey.starts_with(prefixWithDot and idx==nullopt))** Because it actually allows to get object from array, right?

Reply:
> I would suggest to split it to two overload functions , one for object, one for array.

I agree!! thanks 🙇